scapy/contrib/pnio_dcp.py

```diff
@@ -0,0 +1,559 @@
+    ]
+
+    def extract_padding(self, s):
+        return '', s
+
+
+def guess_dcp_block_class(packet, **kargs):
+    """
+    returns the correct dcp block class needed to dissect the current tag
+    if nothing can be found -> dcp base block will be used
+
+    :param packet: the current packet
+    :return: dcp block class
+    """
+
+    option = packet[0]
+    suboption = packet[1]
+
+    def mac_block():
+        return "DCPMACBlock"
+
+    def ip_block():
+        return "DCPIPBlock"
+
+    def manufacturer_specific_block():
+        return "DCPManufacturerSpecificBlock"
+
+    def name_of_station_block():
+        return "DCPNameOfStationBlock"
+
+    def device_id_block():
+        return "DCPDeviceIDBlock"
+
+    def device_role_block():
+        return "DCPDeviceRoleBlock"
+
+    def device_options_block():
+        return "DCPDeviceOptionsBlock"
+
+    def alias_name_block():
+        return "DCPAliasNameBlock"
+
+    def device_instance_block():
+        return "DCPDeviceInstanceBlock"
+
+    def control_response():
+        return "DCPControlBlock"
+
+    # TODO implement the other functions when needed
+
+    class_switch_case = {
+        # IP
+        1:
+            {
+                1: mac_block,
+                2: ip_block
+            },
+        # Device Properties
+        2:
+            {
+                1: manufacturer_specific_block,
+                2: name_of_station_block,
+                3: device_id_block,
+                4: device_role_block,
+                5: device_options_block,
+                6: alias_name_block,
+                7: device_instance_block,
+                8: "OEM Device ID"
+            },
+        # DHCP
+        3:
+            {
+                0x0c: "Host name",
+                0x2b: "Vendor specific",
+                0x36: "Server identifier",
+                0x37: "Parameter request list",
+                0x3c: "Class identifier",
+                0x3d: "DHCP client identifier",
+                0x51: "FQDN, Fully Qualified Domain Name",
+                0x61: "UUID/GUID-based Client",
+                0xff: "Control DHCP for address resolution"
+            },
+        # Control
+        5:
+            {
+                0: "Reserved (0x00)",
+                1: "Start Transaction (0x01)",
+                2: "End Transaction (0x02)",
+                3: "Signal (0x03)",
+                4: control_response,
+                5: "Reset Factory Settings (0x05)",
+                6: "Reset to Factory (0x06)"
+            },
+        # Device Inactive
+        6:
+            {
+                0: "Reserved (0x00)",
+                1: "Device Initiative (0x01)"
+            },
+        # ALL Selector
+        255:
+            {
+                255: "ALL Selector (0xff)"
+            }
+    }
+    try:
+        c = class_switch_case[int(option)][int(suboption)]()
```

Comment:
Are you calling a string here ?

This function is super inconsistent: half are strings, half are useless methods.

Also, please try to remain consistent: why write half of the indexes in hexadecimal, and not the other half (nitpicking) ?